File: src/main/java/plateau/engine/entity/Entity.java
package plateau.engine.entity;

import plateau.engine.world.World;

public class Entity {

	private float x, y, z;
	private float yaw, pitch, roll;
	private float motionX, motionY, motionZ;

	private World world;

	public void update() {
		this.setX(this.getX() + this.getMotionX());
		this.setY(this.getY() + this.getMotionY());
		this.setZ(this.getZ() + this.getMotionZ());

		if (this.getMotionX() != 0) {
			if (this.getMotionX() < 0.00001 && this.getMotionX() > -0.00001) {
				this.setMotionX(0);
			} else {
				this.setMotionX(this.getMotionX() * 0.5f);
			}
		}

		if (this.getMotionZ() != 0) {
			if (this.getMotionZ() < 0.00001 && this.getMotionZ() > -0.00001) {
				this.setMotionZ(0);
			} else {
				this.setMotionZ(this.getMotionZ() * 0.5f);
			}
		}
	}

	public float getX() {
		return this.x;
	}

	public void setX(float x) {
		this.x = x;
	}

	public float getY() {
		return this.y;
	}

	public void setY(float y) {
		this.y = y;
	}

	public float getZ() {
		return this.z;
	}

	public void setZ(float z) {
		this.z = z;
	}

	public float getYaw() {
		return yaw;
	}

	public void setYaw(float yaw) {
		this.yaw = yaw;
	}

	public float getRoll() {
		return roll;
	}

	public void setRoll(float roll) {
		this.roll = roll;
	}

	public float getPitch() {
		return pitch;
	}

	public void setPitch(float pitch) {
		this.pitch = pitch;
	}

	public float getMotionZ() {
		return motionZ;
	}

	public void setMotionZ(float motionZ) {
		this.motionZ = motionZ;
	}

	public float getMotionX() {
		return motionX;
	}

	public void setMotionX(float motionX) {
		this.motionX = motionX;
	}

	public float getMotionY() {
		return motionY;
	}

	public void setMotionY(float motionY) {
		this.motionY = motionY;
	}

	public World getWorld() {
		return world;
	}

	public void setWorld(World world) {
		this.world = world;
	}
}

File: src/entity/Entity.cpp
#include "Entity.h"

Entity::Entity()
{
    x = 0;
    y = 0;
    z = 0;
    yaw = 0;
    pitch = 0;
    roll = 0;
    motionX = 0;
    motionY = 0;
    motionZ = 0;
}

Entity::~Entity()
{

}

void Entity::update()
{
    setX(getX() + getMotionX());
    setY(getY() + getMotionY());
    setZ(getZ() + getMotionZ());

    if (getMotionX() != 0) {
        if (getMotionX() < 0.00001 && getMotionX() > -0.00001) {
            setMotionX(0);
        } else {
            setMotionX(getMotionX() * 0.5f);
        }
    }

    if (getMotionY() != 0) {
        if (getMotionY() < 0.00001 && getMotionY() > -0.00001) {
            setMotionY(0);
        } else {
            setMotionY(getMotionY() * 0.5f);
        }
    }

    if (getMotionZ() != 0) {
        if (getMotionZ() < 0.00001 && getMotionZ() > -0.00001) {
            setMotionZ(0);
        } else {
            setMotionZ(getMotionZ() * 0.5f);
        }
    }
}

float Entity::getX()
{
    return x;
}

float Entity::getY()
{
    return y;
}

float Entity::getZ()
{
    return z;
}

float Entity::getYaw()
{
    return yaw;
}

float Entity::getPitch()
{
    return pitch;
}

float Entity::getRoll()
{
    return roll;
}

float Entity::getMotionX()
{
    return motionX;
}

float Entity::getMotionY()
{
    return motionY;
}

float Entity::getMotionZ()
{
    return motionZ;
}

void Entity::setX(float x1)
{
    x = x1;
}

void Entity::setY(float y1)
{
    y = y1;
}

void Entity::setZ(float z1)
{
    z = z1;
}

void Entity::setYaw(float yaw1)
{
    yaw = yaw1;
}

void Entity::setPitch(float pitch1)
{
    pitch = pitch1;
}

void Entity::setRoll(float roll1)
{
    roll = roll1;
}

void Entity::setMotionX(float motionX1)
{
    motionX = motionX1;
}

void Entity::setMotionY(float motionY1)
{
    motionY = motionY1;
}

void Entity::setMotionZ(float motionZ1)
{
    motionZ = motionZ1;
}
